src/main/java/uk/co/westhawk/snmp/stack/SnmpContextv3Basis.java

```diff
@@ -1179,15 +1260,87 @@ public String toString()
     buffer.append(", contextName=").append(contextName);
     buffer.append(", userName=").append(userName);
     buffer.append(", useAuthentication=").append(useAuthentication);
-    buffer.append(", authenticationProtocol=").append(ProtocolNames[authenticationProtocol]);
+    buffer.append(", authenticationProtocol=").append(PROTOCOL_NAMES[authenticationProtocol]);
     buffer.append(", userAuthenticationPassword=").append(userAuthenticationPassword);
     buffer.append(", usePrivacy=").append(usePrivacy);
-    buffer.append(", privacyProtocol=").append(ProtocolNames[privacyProtocol]);
+    buffer.append(", privacyProtocol=").append(PROTOCOL_NAMES[privacyProtocol]);
     buffer.append(", userPrivacyPassword=").append(userPrivacyPassword);
     buffer.append(", #trapListeners=").append(trapSupport.getListenerCount());
     buffer.append(", #pduListeners=").append(pduSupport.getListenerCount());
     buffer.append("]");
     return buffer.toString();
 }
 
+	/**
+	 * Generates the privacy key based on the authentication protocol.
+	 *
+	 * @param context The SNMP context.
+	 * @param engineId The SNMP engine ID.
+	 * @param authenticationProtocol The authentication protocol.
+	 * @return The generated privacy key.
+	 */
+	protected byte[] generatePrivacyKey(SnmpContextv3Basis context, String engineId, int authenticationProtocol) {
```

Comment:
Remove `SnmpContextv3Basis context` from the method arguments. We are already in the current context.